Dockerfile

```diff
@@ -1,113 +1,44 @@
-        erlang-parsetools erlang-percept erlang-typer erlang-inets \
-        python-mysqldb \
-        imagemagick \
-    ' \
-    && apt-key adv \
-        --keyserver keys.gnupg.net \
-        --recv-keys 434975BD900CCBE4F7EE1B1ED208507CA14F4FCA \
-    && apt-get update \
-    && apt-get install -y $buildDeps $requiredAptPackages --no-install-recommends \
-    && dpkg-reconfigure locales && \
-        locale-gen C.UTF-8 \
-    && /usr/sbin/update-locale LANG=C.UTF-8 \
-    && echo 'en_US.UTF-8 UTF-8' >> /etc/locale.gen \
-    && locale-gen \
-    && cd /tmp \
-    && git clone https://github.com/processone/ejabberd.git \
-        --branch $EJABBERD_BRANCH --single-branch --depth=1 \
-    && cd ejabberd \
-    && chmod +x ./autogen.sh \
-    && ./autogen.sh \
-    && ./configure --enable-user=$EJABBERD_USER \
-        --prefix=/ \
-        --enable-all \
-        --disable-tools \
-        --disable-pam \
-    && make debug=$EJABBERD_DEBUG_MODE \
-    && make install \
-    && mkdir $EJABBERD_HOME/ssl \
-    && mkdir $EJABBERD_HOME/conf \
-    && mkdir $EJABBERD_HOME/backup \
-    && mkdir $EJABBERD_HOME/upload \
-    && mkdir $EJABBERD_HOME/database \
-    && mkdir $EJABBERD_HOME/module_source \
-    && cd $EJABBERD_HOME \
-    && rm -rf /tmp/ejabberd \
-    && rm -rf /etc/ejabberd \
-    && ln -sf $EJABBERD_HOME/conf /etc/ejabberd \
-    && chown -R $EJABBERD_USER: $EJABBERD_HOME \
-    && rm -rf /var/lib/apt/lists/* \
-    && apt-get purge -y --auto-remove $buildDeps
-
-# Wrapper for setting config on disk from environment
-# allows setting things like XMPP domain at runtime
-ADD ./docker/run.sh /sbin/run
-
-# Add run scripts
-ADD ./docker/scripts $EJABBERD_HOME/scripts
-ADD https://raw.githubusercontent.com/rankenstein/ejabberd-auth-mysql/master/auth_mysql.py $EJABBERD_HOME/scripts/lib/auth_mysql.py
-RUN chmod a+rx $EJABBERD_HOME/scripts/lib/auth_mysql.py
-
-# Add config templates
-ADD ./docker/conf /opt/ejabberd/conf
-
-# Continue as user
-USER $EJABBERD_USER
-
-# Set workdir to ejabberd root
-WORKDIR $EJABBERD_HOME
+FROM phusion/baseimage:0.9.22
+
+# Use baseimage-docker's init system.
+CMD ["/sbin/my_init"]
+
+# install dependencies
+RUN apt-get update \
+    && apt-get install -y \
+    make \
+    gcc \
+    expat \
+    libyaml-dev \
+    erlang \
+    erlang-p1-mysql \
+    openssl \
+    elixir \
+    zlib1g-dev \
+    libpng-dev \
+    libexpat-dev \
+    git \
+    automake \
+    libssl-dev \
+    g++
+
+# install ejabberd
+ARG EJABBERD_HOME=/root/ejabberd
+ARG MY_EJABBERD_HOME=/root/my-ejabberd
+
+RUN mkdir $EJABBERD_HOME
+RUN git clone https://github.com/skillz/ejabberd.git --branch development $EJABBERD_HOME
+COPY ejabberd.yml $EJABBERD_HOME/config/ejabberd.yml
+RUN cd $EJABBERD_HOME && ./autogen.sh
+RUN cd $EJABBERD_HOME && ./configure --prefix=$MY_EJABBERD_HOME --enable-elixir
+RUN cd $EJABBERD_HOME && make && make install
+COPY ejabberd.yml $MY_EJABBERD_HOME/etc/ejabberd/ejabberd.yml
```

Comment:
Why is there also a $MY_EJABBERD_HOME ?

Is this to separate the compiled location from the source directory?
If that's the case then the compiled location should be `/opt/ejabberd`